test/util/util.go

```diff
@@ -189,6 +191,65 @@ func wrapErrorWithCertificateStatusCondition(client clientset.CertificateInterfa
 	return pollErr
 }
 
+// WaitCertificateIssuedValid waits for the given Certificate to be
+// 'Ready' and ensures the stored certificate is valid for the specified
+// domains.
+func WaitCertificateIssuedValid(certClient clientset.CertificateInterface, secretClient corecs.SecretInterface, name string, timeout time.Duration) error {
+	return wait.PollImmediate(time.Second, timeout,
+		func() (bool, error) {
+			glog.V(5).Infof("Waiting for Certificate %v to be ready", name)
+			certificate, err := certClient.Get(name, metav1.GetOptions{})
+			if nil != err {
+				return false, fmt.Errorf("error getting Certificate %v: %v", name, err)
+			}
+			isReady := certificate.HasCondition(v1alpha1.CertificateCondition{
+				Type:   v1alpha1.CertificateConditionReady,
+				Status: v1alpha1.ConditionTrue,
+			})
+			if !isReady {
+				return false, nil
+			}
+			glog.Infof("Getting the TLS certificate Secret resource")
+			secret, err := secretClient.Get(certificate.Spec.SecretName, metav1.GetOptions{})
+			if err != nil {
+				return false, err
```

Comment:
If this is a NotFound error, we should instead log a message and `return false, nil` imo. This could be transient in some test cases (i.e. if we delete the Secret resource, the Certificate may still be marked as Ready)